Zend/zend.c

```diff
@@ -1736,3 +1735,49 @@ ZEND_API void zend_map_ptr_extend(size_t last)
 		CG(map_ptr_last) = last;
 	}
 }
+
+static void zend_error_notify_callback_dtor(zend_error_notify_callback *callback)
+{
+	free(callback->name);
+}
+
+int zend_startup_error_notify_callbacks()
+{
+	zend_llist_init(&zend_error_notify_callbacks, sizeof(zend_error_notify_callback), (void (*)(void *)) zend_error_notify_callback_dtor, 1);
+
+	return SUCCESS;
+}
+
+int zend_shutdown_error_notify_callbacks()
+{
+	zend_llist_destroy(&zend_error_notify_callbacks);
+
+	return SUCCESS;
+}
+
+void zend_error_notify_callbacks_apply(llist_apply_func_t callback)
+{
+	zend_llist_apply(&zend_error_notify_callbacks, callback);
+}
+
+void zend_register_error_notify_callback(const char *name, zend_error_notify_cb cb)
+{
+	ZEND_ASSERT(name != NULL);
+
+	zend_error_notify_callback callback;
+
+	callback.name = strdup(name);
+	callback.notify_callback = cb;
+
+	zend_llist_add_element(&zend_error_notify_callbacks, &callback);
+}
+
+static void zend_error_notify_single_callback(zend_error_notify_callback *callback, int type, const char *error_filename, const uint32_t error_lineno, const char *format, va_list args)
```

Comment:
I need to change this still to be: ```static void zend_error_notify_single_callback(zend_error_notify_callback *callback, int num_args, va_list args)```.

Wondering if nested va_args work, loosly this is needed ``error_args = va_args(args, va_list);``